ion-schema/src/isl/isl_type_reference.rs

```diff
@@ -433,55 +435,35 @@ impl IslVariablyOccurringTypeRef {
             true,
         )?;
 
-        let occurs: Range = value
+        let occurs: UsizeRange = value
             .as_struct()
             .and_then(|s| {
                 s.get("occurs")
-                    .map(|r| IslVariablyOccurringTypeRef::occurs_from_ion_element(r, isl_version))
+                    .map(IslVariablyOccurringTypeRef::occurs_from_ion_element)
             })
             .unwrap_or(if constraint_name == "fields" {
-                Ok(Range::optional())
+                Ok(UsizeRange::zero_or_one())
             } else {
-                Ok(Range::required())
+                Ok(UsizeRange::new_single_value(1))
             })?;
 
+        isl_require!(occurs.upper() != &Limit::Closed(0usize) => "Occurs cannot be 0")?;
```

Comment:
🗺️ `isl_require!` is a macro I created along the lines of Kotlin's `require()` function to make it more concise to check ISL syntax. It takes a condition that must be true, and format args for an error message if it's false. You'll see the definition of the macro further down in `result.rs`.

Reply:
Nice